File: easy/ValidParentheses.cpp
#include <iostream>
#include <stack>

using namespace std;

class Solution {
public:
  bool isValid(string s) {
    if (s.size() & 1)
      return false;
    stack<char> st;
    for (auto ch : s) {
      if (ch == '(' || ch == '{' || ch == '[') {
        st.push(ch);
      } else {
        if (st.empty())
          return false;
        char top = st.top();
        if ((ch == ')' && top != '(') || (ch == '}' && top != '{') ||
            (ch == ']' && top != '[')) {
          return false;
        }
        st.pop();
      }
    }
    if (st.empty())
      return true;
    else
      return false;
  }
};

File: 70+/easy/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) <=1 :
            return False

        st = []
        for ch in s:
            if ch == "[" or ch == "{" or ch == "(":
                st.append(ch)
                continue
            if  ((ch == ")" and st   and st[-1] == "(")
                        or (ch=="}" and st and st[-1]=="{")
                        or (ch=="]" and st and st[-1] == "[")
                       ):
                    st.pop()
                    continue

            return False


        return not st


